File: Fibonacci/Barisan_Fibonacci.pas
program BarisanFibonacci;

uses crt;
//Kamus
var
 N, Suku1, Suku2, Fibo, i : integer;
begin
 //Memasukkan banyak suku (N)
 write('Banyak Sukuk (N) : ', );readln(N)
 //Validasi banyak suku (N)
 while( N < 0) or ( N > 20) do
 begin
   write('Banyk suku harus antara 1 - 20, ulangi!!');
   readln;
   gotoxy(1,2); clreol;
   //masukkan banyak suku (N)
   gotoxy(19,1); clreol; readln(N)
   end. //endwhile
   readln;
end

File: BarisanFibonacci.py
# Memasukkan banyak suku N
N = int(input("Masukkan banyak suku N: "))

# Validasi banyak suku
while N < 1 or N > 20:
    print("Banyak suku harus antara 1-20, ulangi!!")
    # Memasukkan banyak suku kembali
    N = int(input("Masukkan banyak suku N: "))

Suku1 = 0
Suku2 = 1

print("Barisan Fibonacci:")
for i in range(N):
    print(Suku1, end=", ")
    Suku1, Suku2 = Suku2, Suku1 + Suku2
